acceptance-tests/src/test/java/hudson/plugins/gradle/MockGeServer.java

```diff
@@ -0,0 +1,118 @@
+            .prefix("scans/publish", c1 -> c1
+                .post("gradle/:pluginVersion/token", this::handleToken)
+                .post("gradle/:pluginVersion/upload", this::handleUpload)
+                .notFound()));
+    }
+
+    @Override
+    protected void after() {
+        mockGeServer.close();
+    }
+
+    private void handleToken(Context ctx) throws Exception {
+        ctx.getRequest().getBody(TEN_MEGABYTES_IN_BYTES).then(request -> {
+            Map<String, Object> requestBody =
+                JSON_OBJECT_MAPPER.readValue(request.getText(), MAP_TYPE_REFERENCE);
+
+            scanTokenRequests.add(
+                new ScanTokenRequest(
+                    (String) requestBody.get("buildToolType"),
+                    (String) requestBody.get("buildToolVersion"),
+                    (String) requestBody.get("buildAgentVersion")
+                ));
+
+            Map<String, String> responseBody =
+                ImmutableMap.of(
+                    "id", PUBLIC_BUILD_SCAN_ID,
+                    "scanUrl", publicBuildScanId(),
+                    "scanUploadUrl", scanUploadUrl(ctx),
+                    "scanUploadToken", DEFAULT_SCAN_UPLOAD_TOKEN
+                );
+
+            ctx.getResponse()
+                .contentType("application/vnd.gradle.scan-ack+json")
+                .send(JSON_WRITER.writeValueAsBytes(responseBody));
+        });
+    }
+
+    private void handleUpload(Context ctx) {
+        ctx.getRequest().getBody(TEN_MEGABYTES_IN_BYTES)
+            .then(__ ->
+                ctx.getResponse()
+                    .contentType("application/vnd.gradle.scan-upload-ack+json")
+                    .send());
+    }
+
+    private String scanUploadUrl(Context ctx) {
+        String pluginVersion = ctx.getPathTokens().get("pluginVersion");
+        return String.format("%sscans/publish/gradle/%s/upload", getAddress(), pluginVersion);
+    }
+
+    public String publicBuildScanId() {
+        return String.format("%ss/%s", getAddress(), PUBLIC_BUILD_SCAN_ID);
+    }
+
+    public URI getAddress() {
+        Preconditions.checkNotNull(mockGeServer, "mockGeServer has not yet been created");
+        return mockGeServer.getAddress();
+    }
+
+    @Nullable
+    public ScanTokenRequest getLastScanTokenRequest() {
+        return Iterables.getLast(scanTokenRequests, null);
+    }
+
+    public static final class ScanTokenRequest {
+
+        public final String toolType;
+        public final String toolVersion;
+        public final String agentVersion;
+
+        private ScanTokenRequest(String toolType, String toolVersion, String agentVersion) {
+            this.toolType = Preconditions.checkNotNull(toolType, "toolType must not be null");
+            this.toolVersion = Preconditions.checkNotNull(toolVersion, "toolVersion must not be null");
+            this.agentVersion = Preconditions.checkNotNull(agentVersion, "agentVersion must not be null");
```

Comment:
Why not use Objects.requireNonNull? It gives you the same syntax and also throws NPE.

Or is it just for consistency that `Preconditions` are used everywhere?

Reply:
Yep, just for consistency. So far all the checks are performed using Guava.